### PR TITLE
Changed PyAttributes type to IndexMap

### PR DESCRIPTION
**Pull request by djrmarques**
Hello!

This is my first pull request, I am very eager to star contributing to this project. 

I believe I have addressed #3607, at least all the tests seem to pass since I made the changes.

Thank you :)

### CODE REVIEW

File: vm/src/builtins/type.rs [hunk 1]
```diff
@@ -39,7 +36,7 @@ pub type PyTypeRef = PyRef<PyType>;
 /// For attributes we do not use a dict, but a hashmap. This is probably
 /// faster, unordered, and only supports strings as keys.
 /// TODO: class attributes should maintain insertion order (use IndexMap here)
```
Comment (youknowone) [inline vm/src/builtins/type.rs:38]: would you also fix the comment here?

**Review by youknowone — CHANGES_REQUESTED** [review]

**Comment by youknowone** [comment]
It looks good in general. I left a few comments not about code

**Comment by djrmarques** [comment]
Hope it's ok now :)

File: vm/src/builtins/type.rs [hunk 2]
```diff
@@ -271,7 +267,7 @@ impl PyType {
     fn dir(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyList {
         let attributes: Vec<PyObjectRef> = zelf
             .get_attributes()
-            .drain()
+            .drain(..)
```
Comment (coolreader18) [inline vm/src/builtins/type.rs:270]: I know `drain()` was already here, but this can actually just be `.into_iter()`.

**Review by youknowone — APPROVED** [review]

**Comment by youknowone** [comment]
I am going to merge it now because drain() is out of boundary of this PR and we should click the run CI button manually for the first contributors.

@djrmarques Thank you for contributing. please feel free to open another PR to chain drain() if you want to do.

**Comment by djrmarques** [comment]
Will do it later probably ☺️

**Pull request merged**